core/src/main/java/org/jdbi/v3/core/mapper/OptionalMapperFactory.java

```diff
@@ -38,28 +42,59 @@
  * </ul>
  */
 class OptionalMapperFactory implements ColumnMapperFactory {
-    private final Map<Class<?>, ColumnMapper<?>> mappers = new IdentityHashMap<>();
-
-    OptionalMapperFactory() {
-        mappers.put(OptionalInt.class, optionalMapper(ResultSet::getInt, OptionalInt::empty, OptionalInt::of));
-        mappers.put(OptionalLong.class, optionalMapper(ResultSet::getLong, OptionalLong::empty, OptionalLong::of));
-        mappers.put(OptionalDouble.class, optionalMapper(ResultSet::getDouble, OptionalDouble::empty, OptionalDouble::of));
-    }
+    private static final Map<Class<?>, OptionalStrategy> STRATEGIES =
+            Collections.unmodifiableMap(Arrays.stream(OptionalStrategy.values())
+                .collect(Collectors.toMap(
+                        OptionalStrategy::optClass,
+                        Function.identity(),
+                        (u, v) -> {
+                            throw new UnsupportedOperationException();
+                        },
+                        IdentityHashMap::new)));
 
     @Override
     public Optional<ColumnMapper<?>> build(Type type, ConfigRegistry config) {
-        Class<?> rawType = getErasedType(type);
+        return Optional.ofNullable(STRATEGIES.get(getErasedType(type)))
+                .map(strategy -> strategy.create(type, config));
+    }
+
+    private enum OptionalStrategy {
```

Comment:
Feels like we're jumping through hoops to shoehorn the primitive mappers and reference mapper into a single type.

Is there anything wrong with putting the reference mapper special case right in the `build()` method? Then `OptionalMapper` could just take a delegate mapper as a constructor parameter.

Reply:
I made it nicer.  Like it better now?

Reply:
:+1: